js/client/props-tree.js

```diff
@@ -0,0 +1,334 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// The property view tree, which groups similar properties for convenience
+
+/*global Tree*/
+
+class PropsTree extends Tree {
+  init() {
+    this.ID_PREFIX = 'pr_';
+
+    const CONTAINER_ID = '#props-tree';
+    const $container = $(CONTAINER_ID);
+    const treeOptions = $.extend(this.DEFAULT_TREE_OPTIONS, {
+      idPrefix: this.ID_PREFIX,
+      quicksearch: true, // Navigate to next node by typing the first letters
+      source: [ ],  // Nothing to show until a node is activated in node-tree
+      keydown: (event, data) => {
+        if (event.keyCode === 13 /* Enter */) {
+          this.onClick(event, data);
+        }
+      },
+      click: this.onClick.bind(this)
+    });
+
+    this.finalize($container, treeOptions);
+  }
+
+  onClick(event, data) {
+    const node = data.node,
+      relationKey = node.data.relationKey;
+    if (relationKey) {
+      window.nodeTree.activate(relationKey);
+      window.nodeTree.tree.setFocus(true);
+    }
+    else {
+      const fnName = node.data.functionName;
+      if (fnName) {
+        this.callAutomationFunction(fnName)
+          .then((value) => {
+            node.setTitle('<a href="#">' + fnName + '()</a> \u2192 ' +
+              JSON.stringify(value));
+          });
+      }
+    }
+  }
+
+  callAutomationFunction(functionName) {
+    const functionCallParams = {
+      type: 'call',
+      key: this.lastKey,
+      functionName,
+      props: []
+    };
+    return window.client.sendMessage(functionCallParams);
+  }
+
+  // Groupings of properties
+  // TODO move to JSON config file
+  getPropMap() {
+    return {
+      'Global': [ 'name', 'nameFrom', 'role', 'state', 'description' ],
+      'Widget': [ 'placeholder', 'value', 'valueForRange', 'minValueForRange',
+        'maxValueForRange', 'textInputType', 'buttonMixed', 'accessKey',
+        'ariaInvalidValue', 'ariaReadonly' ],
+      'HTML': [ 'htmlTag', 'htmlAttributes', 'imageDataUrl' ],
+      'Structure': [ 'posInSet', 'setSize', 'hierarchicalLevel', 'isRootNode' ],
+      'Table': [ 'tableRowCount', 'tableColumnCount', 'tableCellColumnIndex',
+        'tableCellColumnSpan', 'tableCellRowIndex', 'tableCellRowSpan',
+        'tableColumnHeader', 'tableRowHeader' ],
+      'Live region': [ 'liveStatus', 'liveRelevant', 'liveAtomic', 'liveBusy',
+        'containerLiveStatus', 'containerLiveRelevant', 'containerLiveAtomic',
+        'containerLiveBusy' ],
+      'Selection': [ 'anchorOffset', 'anchorAffinity', 'anchorObject',
+        'focusOffset', 'focusAffinity', 'focusObject', 'textSelStart',
+        'textSelEnd' ],
+      'Text': [ 'language', 'wordStarts', 'wordEnds',
```

Comment:
FYI, wordStarts and wordEnds are only on inline text boxes, whereas
language is on any element, and the markers I think are only on
editable text.